api/envoy/extensions/filters/network/mtls_failure_response/v3/mtls_failure_response.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.mtls_failure_response.v3;
+
+import "envoy/type/v3/token_bucket.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.mtls_failure_response.v3";
+option java_outer_classname = "MtlsFailureResponseProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/mtls_failure_response/v3;mtls_failure_responsev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: mTLS Failure Response]
+// [#extension: envoy.filters.network.mtls_failure_response]
+
+// Configuration for the mTLS Failure Response filter. As opposed to returning a SSL failure to a client after failed
+// client cert validation, this filter allows for customization of the response.
+message MtlsFailureResponse {
+  enum ValidationMode {
+    PRESENTED = 0;
+    VALIDATED = 1;
+  }
+
+  enum FailureMode {
+    CLOSE_CONNECTION = 0;
```

Comment:
If you want to just close the connection, that can be done by setting validation in the tls transport socket. Is this option needed?

Reply:
I think the option key is not very clear. The intention as discussed in the issue is to close the connection abruptly without sending a TLS error response. 

It is still possible to inspect the handshake and recognize a client cert was requested even if abruptly closed, but this still might be helpful to limit auto-recon?

Reply:
Oh that's right; I remember that now. Some docs will make this clear.